### ISSUE TITLE
Problem with Evaluation

### ISSUE THREAD

**Issue by duygunuryldz** [shankarp8/knowledge_distillation#4]
Dear authors, 
I have been trying to reproduce your results in the paper. I realized a problem with the evaluation of Entity Inferences dataset. 
To calculate the probability of each label for one sample, probe sentence is forwarded only once with the first label. Then the logits of this pass is used to calculate the probability for all other labels. This approach brings out a problem _when the tokenized lengths of labels are different_.

For example, in ```edit_func.py``` logits for raw model are yielded at line 200:
```  
with torch.no_grad():
        pre_edit_logits = model_raw(**ex).logits
```
where ```ex``` is the probe sentence with the first label and created at line 187: 
```
ex = {}
ex['input_ids'] = batch["edit_inner"][0]['labels']['input_ids'][0].unsqueeze(0)
ex['attention_mask'] = batch["edit_inner"][0]['labels'][ 'attention_mask'][0].unsqueeze(0)
ex['labels'] = batch["edit_inner"][0]['labels']['input_ids'][ 0].unsqueeze(0)
```

Then, at line 224 in the same file probs are calculated for all labels of that sample:
```
with torch.no_grad():
        n_probe_labels = batch['edit_inner'][0]['labels']['input_ids'].size(0)
        pre_edit_dict = []
        for i in range(n_probe_labels):
            label = batch["edit_inner"][0]["labels"]['input_ids'][ i].unsqueeze(0)
            pre_edit_dict.append(get_log_probs( pre_edit_logits, label, shift=True))
```

I believe,  for each label ```pre_edit_logits``` should be obtained by forwarding the probe sentence with that label as follows:
```
with torch.no_grad():
        n_probe_labels = batch['edit_inner'][0]['labels']['input_ids'].size(0)
        pre_edit_dict = []
        for i in range(n_probe_labels):
            ex = {}
            ex['input_ids'] = batch["edit_inner"][0]['labels']['input_ids'][i].unsqueeze(0)
            ex['attention_mask'] = batch["edit_inner"][0]['labels'][ 'attention_mask'][i].unsqueeze(0)
            ex['labels'] = batch["edit_inner"][0]['labels']['input_ids'][i].unsqueeze(0)

            pre_edit_logits = model_raw(**ex).logits
            pre_edit_dict.append(get_log_probs( pre_edit_logits, ex['labels'], shift=True))
``` 

This issue exist also in other functions in ```edit_func.py``` like ```prepend_def_gpt```. You can check the validity of the problem with the 100th sample in ```entity_inferences_augmented.json```

I might also be misundertanding some parts. Please correct me if I am wrong. 
Thanks in advance

**Comment by shankarp8** [comment]
Hey, thanks for this. I'm currently very busy at the moment - I'll try to look into this further in a few weeks. Apologies for the delay.

**Comment by shankarp8** [comment]
I sincerely apologize for this extremely late response; this completely slipped my mind. You are correct, I had inadvertently uploaded versions of run_edit.py and edit_func.py that were only compatible with running ECBD, and not Entity Inferences. I have fixed the errors accordingly in the new versions of the file. Thanks again for pointing this out.
